### PR TITLE
Add new terraform sample - CDN XLB with Backend Bucket

### PR DESCRIPTION
**Pull request by betsy-lichtenberg**
Intending to use this sample on the following page:

https://cloud.google.com/cdn/docs/setting-up-cdn-with-bucket

This PR is an update to https://github.com/GoogleCloudPlatform/magic-modules/pull/5869 and fixes the following issues:

- Adds granular region tags so that we can include subsets of the sample in the various doc sections
- Removes dependencies on the local provider

```release-note:none
```

### CODE REVIEW

**Comment by rileykarson** [comment]
Note, the error here indicates the provider failed to build:

```
==> Checking source code against linters...
google/resource_compute_backend_bucket_generated_test.go:389:36: missing ',' in argument list (typecheck)
# Delete after uncommenting above `source` and `content_type` attributes
                                   ^
make: *** [lint] Error 1
GNUmakefile:26: recipe for target 'lint' failed
```

File: mmv1/products/compute/terraform.yaml [hunk 1]
```diff
@@ -176,6 +176,21 @@ overrides: !ruby/object:Overrides::ResourceOverrides
         primary_resource_id: "image_backend"
         vars:
           backend_bucket_name: "image-backend-bucket"
+      - !ruby/object:Provider::Terraform::Examples
```
Comment (rileykarson) [inline mmv1/products/compute/terraform.yaml:179]: Can we skip the registry docs on this one?

Comment (betsy-lichtenberg) [reply mmv1/products/compute/terraform.yaml:179]: sure

**Comment by betsy-lichtenberg** [comment]
Log says:

"Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object."

I think I need to make the object public.

**Comment by betsy-lichtenberg** [comment]
> Log says:
> 
> "Anonymous caller does not have storage.objects.get access to the Google Cloud Storage object."
> 
> I think I need to make the object public.

Nope, I already did that with this:

```
# make bucket public
resource "google_storage_bucket_iam_member" "default" {
  bucket = google_storage_bucket.default.name
  role   = "roles/storage.objectViewer"
  member = "allUsers"
}
```

So I don't know why I'm getting this error.

**Review by rileykarson — APPROVED** [review]
Looks like there's a conflict that needs to be resolved, otherwise LGTM

**Pull request merged**